.github/workflows/code_checks.yml

```diff
@@ -155,47 +152,46 @@ jobs:
           name: Test Results
           path: log/*.xml
           if-no-files-found: ignore
-      
+
       - name: Upload Test Artifacts
         uses: actions/upload-artifact@v4
-        if: failure()
+        if: ${{ failure() }}
         with:
```

Comment:
Always use ${{ ... }} when calling GitHub Actions expression functions like success(), failure(), in a conditional.